hack/.ci/lib/e2e.sh

```diff
@@ -99,6 +99,10 @@ function gather-artifacts-on-exit {
   cleanup-bg-jobs "${ec}"
 }
 
+function gracefully-shutdown-e2es-on-interrupt {
```

Comment:
SRP: the interrupt is semantically part of the `trap` expression, not this function definition. Suggested rename

```suggestion
function gracefully-shutdown-e2es {
```

+ update to the `trap` call